File: 4284-smallest-stable-index-i/smallest-stable-index-i.cpp
class Solution {
public:
    int firstStableIndex(vector<int>& nums, int k) {
        int left = INT_MIN , right =INT_MAX;
        for(int i = 0 ; i<nums.size(); i++){
            left = max(left, nums[i]);
            for(int j = i ; j< nums.size(); j++){
                right = min(right , nums[j]);
            }
            if(left-right<=k){
                return i;
            }
            right = INT_MAX;
        }
        return -1;
    }
};

File: 4284-smallest-stable-index-i/smallest-stable-index-i.py
class Solution:
    def firstStableIndex(self, nums: list[int], k: int) -> int:
       left = float('-inf')
       for i in range(len(nums)):
        left = max(left , nums[i])

        right = float('inf')
        for j in range(i , len(nums)):
                right = min(right , nums[j])

        if left-right<=k :
             return i
       return -1

        